Neos.Neos/Classes/Controller/Module/Management/WorkspacesController.php

```diff
@@ -792,44 +782,60 @@ protected function computeSiteChanges(Workspace $selectedWorkspace, ContentRepos
                 $documentPathSegments = [];
                 foreach ($ancestors as $ancestor) {
                     $pathSegment = $ancestor->nodeName ?: NodeName::fromString($ancestor->nodeAggregateId->value);
-                    $nodePathSegments[] = $pathSegment;
+                    if (!$this->getNodeType($ancestor)->isOfType(NodeTypeNameFactory::NAME_SITES)) {
+                        $nodePathSegments[] = $pathSegment;
+                    }
                     if ($this->getNodeType($ancestor)->isOfType(NodeTypeNameFactory::NAME_DOCUMENT)) {
                         $documentPathSegments[] = $pathSegment;
                         if (is_null($documentNode)) {
                             $documentNode = $ancestor;
                         }
                         // the site node is the last ancestor of type Document
-                        $siteNode = $documentNode;
+                        // TODO: Check for Neos.Neos:Site instead of regular document after it is introduced
+                        $siteNode = $ancestor;
                     }
                 }
 
                 // Neither $documentNode, $siteNode or its cannot really be null, this is just for type checks;
                 // We should probably throw an exception though
                 if ($documentNode !== null && $siteNode !== null && $siteNode->nodeName) {
                     $siteNodeName = $siteNode->nodeName->value;
-                    $documentPath = implode('/', array_slice(array_map(
+                    $documentPath = implode('/', array_reverse(array_map(
```

Comment:
I'm trying to understand: (how) is this related to the issue?

Reply:
In both places I didn't quite understand why we need the paths in the first place. It's also used by the current JS logic to select and group nodes by document. And in neos 8 that starts at the side node and not with the lowest node.

Reply:
> And in neos 8 that starts at the side node and not with the lowest node

So that's why we change `array_slice` to `array_reverse` right?

Reply:
Yes